src/phoenix/experimental/evals/templates/template.py

```diff
@@ -16,101 +15,127 @@
-        else:
-            raise ValueError("delimiters must only contain 2 items in the list")
-
-    def _validate(self) -> None:
-        # Validate that for every open delimiter, we have the corresponding closing one
-        start_count = self.text.count(self._start_delim)
-        end_count = self.text.count(self._end_delim)
-        if start_count != end_count:
-            raise ValueError(
-                f"text poorly formatted. Found {start_count} instances of delimiter "
-                f"{self._start_delim} and {end_count} instances of {self._end_delim}. "
-                "They must be equal to be correctly paired."
-            )
+    def _parse_variables(self, text: str) -> List[str]:
+        pattern = re.escape(self._start_delim) + "(.*?)" + re.escape(self._end_delim)
+        variables = re.findall(pattern, text)
+        return variables
 
-    def _parse_variables(self) -> List[str]:
-        variables = []
-        formatter = Formatter()
-
-        text = self.text
-
-        # Example of this could be a template like: My name is ::name::
-        if self._start_delim == self._end_delim:
-            delim_length = len(self._start_delim)
-            delim_count = text.count(self._start_delim)
-            while delim_count > 0:
-                left_index = text.find(self._start_delim)
-                right_index = text[left_index + delim_length :].find(self._start_delim)
-                text = (
-                    text[0:left_index]
-                    + DEFAULT_START_DELIM
-                    + text[left_index + delim_length : left_index + delim_length + right_index]
-                    + DEFAULT_END_DELIM
-                    + text[left_index + 2 * delim_length + right_index :]
-                )
-                delim_count = text.count(self._start_delim)
-        else:
-            if self._start_delim != "{":
-                text = text.replace(self._start_delim, DEFAULT_START_DELIM)
-            if self._end_delim != "{":
-                text = text.replace(self._end_delim, DEFAULT_END_DELIM)
+    @abstractmethod
+    def parse_label(self, raw_string: str) -> str:
+        ...
 
-        for _, variable_name, _, _ in formatter.parse(text):
-            if variable_name:
-                variables.append(variable_name)
 
-        return variables
+class UserTemplate(ClassificationTemplate):
+    text: str
+    variables: List[str]
+
+    def __init__(
+        self,
+        text: str,
+        delimiters: Tuple[str, str] = (DEFAULT_START_DELIM, DEFAULT_END_DELIM),
+    ):
+        self.text = text
+        self._start_delim, self._end_delim = delimiters
+        self.variables = self._parse_variables(self.text)
+
+    def prompt(self, options: Optional[PromptOptions]) -> str:
+        return self.text
+
+    def parse_label(self, raw_string: str) -> str:
+        raise TypeError("UserTemplates do not support parsing complex outputs.")
+
+
+class PhoenixTemplate(ClassificationTemplate):
```

Comment:
I also don't follow the need for this either - maybe we can discuss this - it feels like we are using classes to extend the behavior but in a non-intuitive way - like now I need to know what's the difference between a user and Phoenix template - it might be a matter of naming mainly but I think this is not entirelyintuitive.

Reply:
Is this intended to be a `ClassificationWithExplanationTemplate`?